### DISCUSSION TITLE
How to write more readable queries? [Question]

### DISCUSSION THREAD

**Discussion by PedroDivergence** [sidorares/node-mysql2#1542] [Q&A] [answered]
It's almost impossible to read a long query string without code indentation. 

`SELECT  T_1.field_with_big_name, T_1.another_big_field_name, T_2.again_another_big_field FROM MY_TABLE_NAME_NUMBER_1 as T_1 WHERE ` T_1.another_big_field_name` = "ABCDEF JOIN ......."`

I would like to have more indentation; something like this:
```
SELECT
  T_1.field_with_big_name,
   T_1.another_big_field_name, 
   T_2.again_another_big_field 
FROM 
   MY_TABLE_NAME_NUMBER_1 as T_1 
WHERE 
` T_1.another_big_field_name` = "ABCDEF 
JOIN
 ......."`
 ```
 
Would that be possible?

Thanks.

**Comment by sidorares** [comment, marked as answer]
I guess the easiest way to keep it formatted in the source code is to use string literals:

```js
const [rows] = await connection.query(`
SELECT
  T_1.field_with_big_name,
   T_1.another_big_field_name, 
   T_2.again_another_big_field 
FROM 
   MY_TABLE_NAME_NUMBER_1 as T_1 
WHERE 
  \`T_1.another_big_field_name\` = "ABCDEF"
JOIN
`);
```

if you have lots of backticks in the query text this might still look a bit noisy ( [SO: how to escape backicks in string literal](https://stackoverflow.com/questions/35803959/template-literals-with-nested-backticks-in-es6) ). Possible other solutions: have a set of plain text files ( with `.sql` extension ) and read them at startup into a dictionary `{ file_name: string_with_sql_content }`. As a bonus you'll get syntax highlighting. The code will look like this:

```js
// assuming queries object contains name:query pairs loaded for example from files in queries/[name].sql
const [rows] = await connection.execute(queries.my_big_query_with_parameters, [parameter]);
```

**Reply by PedroDivergence** [reply]
@sidorares Thank you.
